### PR TITLE
store dapp instruction data inside multisig op

### PR DESCRIPTION
**Pull request by bholzman**
## Description
Made multisig op data large enough to hold dapp tx data too, and eliminate multisig data account.

## Motivation and Context
 Saves some more space in dapp finalization because multisig data account does not need to be passed in.

## How Has This Been Tested?
All existing tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Bug fix breaking (fix that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] New feature breaking (change which adds functionality and causes existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/handlers/dapp_transaction_handler.rs [hunk 1]
```diff
@@ -115,29 +108,26 @@ pub fn supply_instructions(
...
                     .get(usize::from(index - starting_index))
                     .unwrap(),
             )?;
         }
 
-        let params_hash = if multisig_data.all_instructions_supplied() {
-            Some(multisig_data.hash(&multisig_op)?)
-        } else {
-            None
+        let params_hash = match multisig_op.dapp_tx_data {
```
Comment (aonyshchenko-strike) [inline src/handlers/dapp_transaction_handler.rs:120]: could be simplified into something like:

```
let params_hash = multisig_op.dapp_tx_data.flat_map(|data| if data.all_instructions_supplied() {
  Some(multisig_op.dapp_hash()?)
} else {
  None
})
```

**Review by aonyshchenko-strike — APPROVED** [review]
Looks great!

**Review by bflood-strike — APPROVED** [review]
Looks great

File: src/handlers/dapp_transaction_handler.rs (hunk 1 above)
Comment (bholzman) [reply src/handlers/dapp_transaction_handler.rs]: was a little more involved to get working, but still an improvement, thanks

**Review by aonyshchenko-strike — DISMISSED** [review]

**Review by bflood-strike — APPROVED** [review]

**Pull request merged**